docs/addons/amp-addon.md

```diff
@@ -0,0 +1,123 @@
+# Amazon Managed Service for Prometheus (AMP)) Amazon EKS Add-on
+
+Amazon Managed Service for Prometheus is a Prometheus-compatible monitoring and alerting service that makes it easy to monitor containerized applications and infrastructure at scale. This Addon deploys an AWS Distro for OpenTelemetry (ADOT) Collector for Amazon Managed Service for Prometheus (AMP) which receives OTLP metrics from the application and Prometheus metrics scraped from pods on the cluster and remote writes the metrics to AMP remote write endpoint. You can change the mode to Daemonset, StatefulSet, and Sidecar depending on your deployment strategy.
+
+This driver is not automatically installed when you first create a cluster, it must be added to the cluster in order to setup AMP for remote write of metrics.
+
+For more information on the driver, please review the [user guide](https://docs.aws.amazon.com/eks/latest/userguide/opentelemetry.html).
+
+## Prerequisites
+- `Certificate-Manager` Blueprints Addon addon.
+- `ADOT` EKS Blueprints Addon.
+
+## Usage
+
+This Addon can used with four different patterns :
+
+Pattern # 1 : Simple and Easy - Using all default property values.
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.AmpAddOn();
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOn)
+  .build(app, 'my-stack-name');
+```
+
+Pattern # 2 : Overriding property values for Name and Tags.
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.AmpAddOn({
+    workspaceName: 'sample-AMP-Workspace',
+    workspaceTagKey: 'Environment',
+    workspaceTagValue: 'Dev',
+    deploymentMode: DeploymentMode.DEPLOYMENT
+})
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOn)
+  .build(app, 'my-stack-name');
+```
+Pattern # 3 : Passing on AMP Remote Write Endpoint URL of an existing AMP workspace to be used to remote write metrics.
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.AmpAddOn({
+    prometheusRemoteWriteURL: 'https://aps-workspaces.us-west-2.amazonaws.com/workspaces/ws-e859f589-7eed-43c1-a82b-58f44119f17d/api/v1/remote_write',
+    deploymentMode: DeploymentMode.DEPLOYMENT
+})
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOn)
+  .build(app, 'my-stack-name');
+```
+
+Pattern # 4 : Overriding Property values for Different Deployment Modes. Supported Ones - `deployment`, `daemonset`, `statefulset`, `sidecar`.
```

Comment:
Nice!

Reply:
Thankyou! Added more details to the docs to make this more clear. All required details are added.